provider/datagen/src/transform/cldr/currency/mod.rs

```diff
@@ -0,0 +1,357 @@
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use icu_locid::LanguageIdentifier;
+use icu_properties::sets::load_for_general_category_group;
+use icu_properties::GeneralCategoryGroup;
+use icu_provider::DataProvider;
+use tinystr::UnvalidatedTinyAsciiStr;
+use zerovec::maps::MutableZeroVecLike;
+use zerovec::VarZeroVec;
+use zerovec::ZeroMap;
+
+use crate::transform::cldr::cldr_serde;
+use crate::DatagenProvider;
+use icu_provider::datagen::IterableDataProvider;
+use icu_provider::prelude::*;
+use icu_singlenumberformatter::provider::*;
+use std::collections::HashMap;
+use tinystr::tinystr;
+
+fn which_currency_pattern(
+    provider: &DatagenProvider,
+    pattern: String,
+    place_holder: String,
+) -> Result<bool, DataError> {
+    // If the pattern is empty,
+    // then return a letter (L) is close to the number.
+    // Therefore, we use the standard_alpha_next_to_number pattern.
+    if place_holder.is_empty() {
+        return Ok(false);
+    }
+
+    let currency_sign = '¤';
+    let currency_sign_ind = pattern.find(currency_sign).unwrap();
+    let first_num_ind = pattern
+        .chars()
+        .enumerate()
+        .find(|(_, c)| c == &'0' || c == &'#')
+        .unwrap()
+        .0;
+    let last_num_ind = pattern.len()
+        - pattern
+            .chars()
+            .rev()
+            .enumerate()
+            .find(|(_, c)| c == &'0' || c == &'#')
+            .unwrap()
+            .0
+        - 1;
+
+    let letters_set = match load_for_general_category_group(provider, GeneralCategoryGroup::Letter)
+    {
+        Ok(letters_set) => letters_set,
+        Err(icu_properties::PropertiesError::PropDataLoad(e)) => {
+            return Err(e);
+        }
+        Err(_) => unreachable!(),
+    };
+
+    let right_letter_test_position = {
+        if currency_sign_ind < first_num_ind && currency_sign_ind < last_num_ind {
+            true
+        } else if currency_sign_ind > first_num_ind && currency_sign_ind > last_num_ind {
```

Comment:
iiuc `last_num_ind > first_num_ind`, so you only have to check

```suggestion
        } else if currency_sign_ind > last_num_ind {
```